apps/console/src/features/roles/components/edit-role/edit-role-groups-updated.tsx

```diff
@@ -0,0 +1,547 @@
+import { addAlert } from "@wso2is/core/store";
+import { EmphasizedSegment, EmptyPlaceholder, Heading } from "@wso2is/react-components";
+import { AxiosError } from "axios";
+import debounce, { DebouncedFunc } from "lodash-es/debounce";
+import isEmpty from "lodash-es/isEmpty";
+import React, {
+    FunctionComponent,
+    HTMLAttributes,
+    ReactElement,
+    SyntheticEvent,
+    useCallback,
+    useEffect,
+    useState
+} from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch } from "react-redux";
+import { Dispatch } from "redux";
+import { AutoCompleteRenderOption } from "./edit-role-common/auto-complete-render-option";
+import { RenderChip } from "./edit-role-common/render-chip";
+import { updateResources } from "../../../core/api/bulk-operations";
+import { getEmptyPlaceholderIllustrations } from "../../../core/configs/ui";
+import { 
+    GroupsInterface, 
+    PatchBulkGroupDataInterface, 
+    PatchGroupAddOpInterface, 
+    PatchGroupOpInterface, 
+    PatchGroupRemoveOpInterface 
+} from "../../../groups";
+import { useGroupList } from "../../../groups/api/groups";
+import { UserBasicInterface } from "../../../users/models";
+import { RoleConstants } from "../../constants";
+import { AssignedInterface } from "../../models/roles";
+import { RoleManagementUtils } from "../../utils";
+
+type RoleGroupsPropsInterface = IdentifiableComponentInterface & AssignedInterface;
+
+export const RoleGroupsList: FunctionComponent<RoleGroupsPropsInterface> = (
+    props: RoleGroupsPropsInterface
+): ReactElement => {
+
+    const {
+        role,
+        onRoleUpdate,
+        isReadOnly
+    } = props;
+
+    const { t } = useTranslation();
+    const dispatch: Dispatch = useDispatch();
+
+    const [ groupSearchValue, setGroupSearchValue ] = useState<string>(undefined);
+    const [ isGroupSearchLoading, setGroupSearchLoading ] = useState<boolean>(false);
+    const [ groupsOptions, setGroupsOptions ] = useState<GroupsInterface[]>([]);
+    const [ selectedGroupsOptions, setSelectedGroupsOptions ] = useState<GroupsInterface[]>(undefined);
+    const [ initialSelectedGroupsOptions, setInitialSelectedGroupsOptions ] = useState<GroupsInterface[]>(undefined);
+    const [ removedGroupsOptions, setRemovedGroupsOptions ] = useState<GroupsInterface[]>(undefined);
+    const [ isSubmitting, setIsSubmitting ] = useState<boolean>(false);
+    const [ activeOption, setActiveOption ] = useState<GroupsInterface|UserBasicInterface>(undefined);
+    const [ showEmptyRolesListPlaceholder, setShowEmptyRolesListPlaceholder ] = useState<boolean>(false);
+
+    const {
+        data: originalGroupList,
+        isLoading: isGroupListFetchRequestLoading,
+        error: groupListFetchRequestError,
+        mutate: mutateGroupListFetchRequest
+    } = useGroupList(null, null, groupSearchValue ? `displayName co ${ groupSearchValue }` : null);
+
+    /**
+     * set initial selected groups options
```

Comment:
Check the other places 

```suggestion
     * Set initial selected groups options
```

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/4212/commits/c57c7b953b186b80c95a389a9c46805020b584fa